Change label color for course info and metadata labels

noisecapella opened the pull request:
#### Pre-Flight checklist

- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #25 

#### What's this PR do?
Updates text color for course info and text labels

#### How should this be manually tested?
View the course info section of the course home page, and the course info section of the side of a section page. The text should be darker than before

#### Screenshots (if appropriate)
![Screenshot from 2021-05-19 16-54-44](https://user-images.githubusercontent.com/863262/118883159-f0ea8680-b8c2-11eb-8a14-60fd47c4cf6a.png)

![Screenshot from 2021-05-19 16-54-27](https://user-images.githubusercontent.com/863262/118883172-f34ce080-b8c2-11eb-8a9b-c4c25228d354.png)

aliceriot commented on base-theme/assets/css/variables.scss:
```diff
@@ -24,6 +24,7 @@ $card-background: #fff;
 $font-gray: #888;
 $podcast-border-gray: #9c9d9e;
 $medium-gray: #c2c0bf;
+$label-gray: #3d3d3d;
```
ayy what a mess of colors we have here - this file defines no less than 22 different gray colors. What do you think of using this opportunity to get it down to a more reasonable level? In my mind there should probably only be 3 or so: a font gray, a background / unfocused font gray, and a border color.

maybe at least while we're in here we could delete the ones we're not using at all right now? it looks like `$bg-gray` isn't used, possibly others

noisecapella replied:
@abdulkdawson Any objections to me merging some grays together until we get down to 3?

noisecapella replied:
cc @abdulkdawson. Do we have a palette that we are using?

abdulkdawson replied:
@noisecapella I have no objections at all.

abdulkdawson replied:
@noisecapella These are only shades of grey that I've been using. #3D3D3D, #F5F5F5 and #9C9D9E

aliceriot reviewed (changes requested):
one question

noisecapella commented:
@aliceriot I got it down to 5 colors: the three @abdulkdawson mentioned, and two with transparency. What do you think?

In general I renamed the variables to `light-gray`, `medium-gray`, and `dark-gray`, and removed the other gray variables (and `$card-background` which was an alias for white). For the most part I just replaced the gray with `$medium-gray` since it seemed closest to the before color, except for `text-box-background` which seemed closer to `light-gray`, and `podcast-border-gray` which was closer to `dark-gray`. The original `dark-gray` was also closer to the current `light-gray` ironically so there are some counterintuitive changes there. I also tried to clean up the CSS classes.
